tools/integration-test/src/tests/channel_upgrade/upgrade_handshake_steps.rs

```diff
@@ -1204,6 +1213,222 @@ impl BinaryChannelTest for ChannelUpgradeHandshakeFlushPackets {
+        // Initialise a new upgrade handshake on chain B before ChannelEnd A moves to `OPEN`
+
+        let pre_upgrade_2_ordering = channel_end_a.ordering;
+        let pre_upgrade_2_connection_hops_b = channel_end_b.connection_hops.clone();
+
+        let post_upgrade_2_version = Version::ics20();
+
+        info!("Will initialise on chain B upgrade handshake with governance proposal...");
+
+        chains.node_b.chain_driver().initialise_channel_upgrade(
+            channel.dst_port_id().as_str(),
+            channel.dst_channel_id().unwrap().as_str(),
+            pre_upgrade_2_ordering.as_str(),
+            pre_upgrade_2_connection_hops_b.first().unwrap().as_str(),
+            &serde_json::to_string(&post_upgrade_2_version.0).unwrap(),
+            chains.handle_b().get_signer().unwrap().as_ref(),
+            "1",
+        )?;
+
+        info!("Check that the step ChanUpgradeInit was correctly executed...");
+
+        channel_end_b = chains
+            .handle_b
+            .query_channel(
+                QueryChannelRequest {
+                    port_id: channels.port_b.0.clone(),
+                    channel_id: channels.channel_id_b.0.clone(),
+                    height: QueryHeight::Latest,
+                },
+                IncludeProof::No,
+            )
+            .map(|(channel_end, _)| channel_end)
+            .map_err(|e| eyre!("Error querying ChannelEnd B: {e}"))?;
+
+        // upgrade sequence should have been incremented
+        let upgrade_sequence_b = Sequence::from(2);
+        if !channel_end_b.upgrade_sequence.eq(&upgrade_sequence_b) {
+            return Err(Error::generic(eyre!(
+                "expected channel end B upgrade sequence to be `{}`, but it is instead `{}`",
+                upgrade_sequence_b,
+                channel_end_b.upgrade_sequence
+            )));
+        }
+
+        // Finish upgrade 1 on ChannelEnd A
+
+        info!("Will run ChanUpgradeOpen step...");
+
+        channel.flipped().build_chan_upgrade_open_and_send()?;
+
+        info!("Check that the step ChanUpgradeOpen was correctly executed...");
+
+        channel_end_a = chains
+            .handle_a
+            .query_channel(
+                QueryChannelRequest {
+                    port_id: channels.port_a.0.clone(),
+                    channel_id: channels.channel_id_a.0.clone(),
+                    height: QueryHeight::Latest,
+                },
+                IncludeProof::No,
+            )
+            .map(|(channel_end, _)| channel_end)
+            .map_err(|e| eyre!("Error querying ChannelEnd A: {e}"))?;
+
+        let state_a = ChannelState::Open(UpgradeState::NotUpgrading);
+        if !channel_end_a.state_matches(&state_a) {
+            return Err(Error::generic(eyre!(
+                "expected channel end A state to be `{}`, but is instead `{}`",
+                state_a,
+                channel_end_a.state()
+            )));
+        }
```

Comment:
Shouldn't we check that the complete upgrade was successful instead of only checking the state?